plugins/module_utils/network/sonic/config/ntp/ntp.py

```diff
@@ -0,0 +1,550 @@
+                        server.pop('maxpoll')
+
+    def get_create_requests(self, configs, have):
+
+        requests = []
+
+        enable_auth_config = configs.get('enable_ntp_auth', None)
+        if enable_auth_config is not None:
+            enable_auth_request = self.get_create_enable_ntp_auth_requests(enable_auth_config, have)
+            if enable_auth_request:
+                requests.extend(enable_auth_request)
+
+        src_intf_config = configs.get('source_interfaces', None)
+        if src_intf_config:
+            src_intf_request = self.get_create_source_interface_requests(src_intf_config, have)
+            if src_intf_request:
+                requests.extend(src_intf_request)
+
+        keys_config = configs.get('ntp_keys', None)
+        if keys_config:
+            keys_request = self.get_create_keys_requests(keys_config, have)
+            if keys_request:
+                requests.extend(keys_request)
+
+        servers_config = configs.get('servers', None)
+        if servers_config:
+            servers_request = self.get_create_servers_requests(servers_config, have)
+            if servers_request:
+                requests.extend(servers_request)
+
+        trusted_key_config = configs.get('trusted_keys', None)
+        if trusted_key_config:
+            trusted_key_request = self.get_create_trusted_key_requests(trusted_key_config, have)
+            if trusted_key_request:
+                requests.extend(trusted_key_request)
+
+        vrf_config = configs.get('vrf', None)
+        if vrf_config:
+            vrf_request = self.get_create_vrf_requests(vrf_config, have)
+            if vrf_request:
+                requests.extend(vrf_request)
+
+        return requests
+
+    def preprocess_delete_commands(self, commands, have):
+
+        new_commands = {}
+
+        enable_auth_command = commands.get('enable_ntp_auth', None)
+        enable_auth_have = have.get('enable_ntp_auth', None)
+        if enable_auth_command is not None and enable_auth_have is not None and \
+           (enable_auth_command or enable_auth_have):
+            new_commands['enable_ntp_auth'] = True
+
+        keys_command = commands.get('ntp_keys', None)
+        if keys_command:
+            new_commands['ntp_keys'] = commands['ntp_keys']
+
+        servers_command = commands.get('servers', None)
+        if servers_command:
+            new_commands['servers'] = commands['servers']
+
+        src_intf_command = commands.get('source_interfaces', None)
+        if src_intf_command:
+            new_commands['source_interfaces'] = commands['source_interfaces']
+
+        trusted_key_command = commands.get('trusted_keys', None)
+        if trusted_key_command:
+            new_commands['trusted_keys'] = commands['trusted_keys']
+
+        vrf_command = commands.get('vrf', None)
+        if vrf_command:
+            new_commands['vrf'] = commands['vrf']
+
+        return new_commands
+
+    def get_delete_requests(self, configs, have):
```

Comment:
None of the functions called from this function currently make use of the "have" parameter. If this is not changed (to eliminate the "preprocess_delete_commands" function), the "have" parameter can be deleted for this function and also from the functions that this one calls.

Reply:
Done

Reply:
Pending: We need to revisit the reason for the preprocess_delete_commands function. (See the related comments above.)

Reply:
Removed preprocess_delete_commands function.

Reply:
The current parameter usage after the most recent push to this PR looks good.